ce/ce/session.ts

```diff
@@ -329,39 +214,30 @@ export class Session {
   async deactivate() {
     const previous = this.environment[undo];
     if (previous && this.postscriptFile) {
-      const deactivationDataFile = this.parseUri(previous);
+      const deactivationDataFile = this.fileSystem.file(previous);
       if (deactivationDataFile.scheme === 'file' && await deactivationDataFile.exists()) {
```

Comment:
```suggestion
      if (await deactivationDataFile.exists()) {
```
This is also the usual exists+read antipattern